pex/bin/pex.py

```diff
@@ -291,6 +291,17 @@ def configure_clp_pex_options(parser):
            'No value (alias for prefer, for backwards compatibility). '
            '[Default: %default]')
 
+  group.add_option(
+      '--reproducible', '--not-reproducible',
+      dest='reproducible',
+      default=True,
```

Comment:
Premature optimization may be the root of all evil, but the optimization is established with a ~9 year precedent. I think it is unwise to change this default of long standing - there are real non-Pants pex customers out there.

Reply:
> but the optimization is established with a ~9 year precedent. 

Agreed. I support sticking with precedent and defaulting to not reproducible, particularly because `--reproducible` means either they will have to define `SOURCE_DATE_EPOCH` or we hardcode it with some arbritary value, per https://github.com/pantsbuild/pex/pull/718/files#r279998538.

cc @illicitonion today, who was curious if it would be feasible to default to `--reproducible`, if you have thoughts on this all.